src/software/geom/algorithms/step_along_perimeter.cpp

```diff
@@ -33,15 +33,24 @@ Point stepAlongPerimeter(const Polygon& polygon, const Point& start,
     double perimeter          = polygon.perimeter();
     bool is_counter_clockwise = travel_distance < 0;
     travel_distance           = std::fmod(std::abs(travel_distance), perimeter);
-    if (is_counter_clockwise)
+
+    if (is_counter_clockwise && polygon.isClockwise())
     {
         travel_distance = perimeter - travel_distance;
     }
```

Comment:
Should we account for when the polygon is counter-clockwise and `is_counter_clockwise` is false?

Reply:
we should